### PR TITLE
feat: :sparkles: ClearML training loss logging

### PR DESCRIPTION
**Pull request by odulcy-mindee**

### CODE REVIEW

File: references/detection/train_pytorch.py [hunk 1]
```diff
@@ -377,32 +404,26 @@ def main(args):
...
+
     # Create loss queue
     min_loss = np.inf
     if args.early_stop:
         early_stopper = EarlyStopper(patience=args.early_stop_epochs, min_delta=args.early_stop_delta)
 
     # Training loop
     for epoch in range(args.epochs):
-        fit_one_epoch(model, train_loader, batch_transforms, optimizer, scheduler, amp=args.amp)
+        fit_one_epoch(model, train_loader, batch_transforms, optimizer, scheduler, amp=args.amp, clearml=args.clearml)
```
Comment (felixdittrich92) [inline references/detection/train_pytorch.py:426]: clearml_log instead :)

**Review by felixdittrich92 — CHANGES_REQUESTED** [review]

File: references/detection/train_pytorch.py (hunk 1 above)
Comment (odulcy-mindee) [reply references/detection/train_pytorch.py:426]: Good catch, thanks

**Review by felixdittrich92 — APPROVED** [review]
Thanks :)

**Comment by kuraga** [comment]
Sorry, weren't there plans to make logging more universal?
(`clearml_log` argument name doesn't have a place for other loggers...)

**Comment by felixdittrich92** [comment]
Hi @kuraga 👋🏼,

What do you mean with other loggers ? 
Sry I'm a bit out of scope with the Clearml topic (still using W&B on my own 😅)
CC @odulcy-mindee

**Comment by felixdittrich92** [comment]
> @felixdittrich92 Hm... Is `W&B` exactly the answer? 😄
> 
> I like a lightweight [lazyscribe](https://github.com/lazyscribe/lazyscribe) BTW...
> 
> I don't ask "to implement other loggers" but about some abstract layer...

Ah got it I was thinking the question is related to something with ClearML :sweat_smile: For me personally W&B is still the answer xD But yeah everyone what they prefer

Mh yes I understood the request ..the thing is that the training scripts / references folder is more a "collection" of scripts everyone can modify to there own needs which has some advantages but on the other side also disadvantages (for example adding globally avaialble features shared between the different training tasks like logging, early stopping, and so on)

with an `Trainer` object like you know from the `transformers` lib for example it would be much easier to implement such features but on the other hand it would wrap everything, reduce the transparency and make it much harder for (experienced) users to transform the training process to there own needs :sweat_smile:

**Comment by kuraga** [comment]
> the training scripts / references folder is more a "collection" of scripts

Aaahh, didn't catch it. My fault. Training of docTR is Mindee's inner job at most. Thanks, @felixdittrich92 :)

P.S. Add READMEs to `/references`, etc. :)

**Comment by felixdittrich92** [comment]
Mh not directly we use the same scripts for pre-training all the models but users can use it also for fine-tuning on there own needs :)

The inner task specific folders have already README's :)
For example: https://github.com/mindee/doctr/tree/main/references/recognition

**Comment by felixdittrich92** [comment]
And the corresponding doc: https://mindee.github.io/doctr/latest/using_doctr/custom_models_training.html

**Comment by kuraga** [comment]
Well, if the mechanism is public I'll illustrate my

> I don't ask "to implement other loggers" but about some abstract layer...

: the `clearml_log: bool` argument could be `logger: logging.Logger` :)

**Comment by kuraga** [comment]
Update: hhm, now [I see](https://github.com/clearml/clearml/blob/v1.17.0/clearml/logger.py#L37) `clearml.Logger` **does not** inherit `logging.Logger`...

**Comment by ThanhNX0401** [comment]
Is there any reason why docTR don't output the Training loss log to WB? How can i tell if the model is overfit or underfit?

**Comment by felixdittrich92** [comment]
> Is there any reason why docTR don't output the Training loss log to WB? How can i tell if the model is overfit or underfit?

Oh you are right we should do this for every train script the same way it's done with the `clearml_log` passed to the train function. Would you like to open a PR ?

**Comment by felixdittrich92** [comment]
@ThanhNX0401 Fixed now on main branch 😊
